<launch>
  <!-- Load multiple TurtleBot3 in TurtleBot3 House -->
  <include file="$(find turtlebot3_gazebo)/launch/multi_turtlebot3.launch"/>

  <!-- Launch SLAM for each TurtleBot3 -->
  <group ns="tb3_0">
    <include file="$(find turtlebot3_slam)/launch/turtlebot3_gmapping.launch">
      <arg name="set_base_frame" value="tb3_0/base_footprint"/>
      <arg name="set_odom_frame" value="tb3_0/odom"/>
      <arg name="set_map_frame" value="tb3_0/map"/>
    </include>
  </group>

  <!-- Launch SLAM Move Base for robot 0 -->
  <include file="$(find turtlebot_navigation_and_mapping)/launch/custom_move_base.launch">
     <arg name="robot_namespace" value="tb3_0"/>
  </include>

  <group ns="tb3_1">
    <include file="$(find turtlebot3_slam)/launch/turtlebot3_gmapping.launch">
      <arg name="set_base_frame" value="tb3_1/base_footprint"/>
      <arg name="set_odom_frame" value="tb3_1/odom"/>
      <arg name="set_map_frame" value="tb3_1/map"/>
    </include>
  </group>

  <!-- Launch SLAM Move Base for robot 1 -->
  <include file="$(find turtlebot_navigation_and_mapping)/launch/custom_move_base.launch">
     <arg name="robot_namespace" value="tb3_1"/>
  </include>




  <group ns="tb3_2">
    <include file="$(find turtlebot3_slam)/launch/turtlebot3_gmapping.launch">
      <arg name="set_base_frame" value="tb3_2/base_footprint"/>
      <arg name="set_odom_frame" value="tb3_2/odom"/>
      <arg name="set_map_frame" value="tb3_2/map"/>
    </include>
  </group>

  <!-- Launch SLAM Move Base for robot 2 -->
  <include file="$(find turtlebot_navigation_and_mapping)/launch/custom_move_base.launch">
     <arg name="robot_namespace" value="tb3_2"/>
  </include>




  <!-- Merge map data from each TurtleBot3 -->
  <include file="$(find turtlebot3_gazebo)/launch/multi_map_merge.launch"/>

  <!-- Launch RViz -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find turtlebot3_gazebo)/rviz/multi_turtlebot3_slam.rviz"/>

  <!-- Run your Python script for each TurtleBot3 -->
<!-- <node name="multi_robot_control" pkg="turtlebot_navigation_and_mapping" type="multi_explore.py"/> -->

</launch>
